cg/meta/workflow/fluffy.py

```diff
@@ -139,31 +191,58 @@ def add_concentrations_to_samplesheet(
         Adds columns Library_nM, SequencingDate, Exclude and populates with orderform values
         """
 
-        samplesheet_df = pd.read_csv(
-            samplesheet_housekeeper_path, index_col=None, header=0, skiprows=4
+        samplesheet_df = self.read_samplesheet_data(
+            samplesheet_housekeeper_path=samplesheet_housekeeper_path
         )
+
         LOG.info(samplesheet_df)
         sample_id_column_alias = (
             "Sample_ID" if "Sample_ID" in samplesheet_df.columns else "SampleID"
         )
         sample_project_column_alias = (
             "Sample_Project" if "Sample_Project" in samplesheet_df.columns else "Project"
         )
-        samplesheet_df["SampleName"] = samplesheet_df[sample_id_column_alias].apply(
-            lambda x: self.get_sample_name_from_lims_id(lims_id=x)
+
```

Comment:
something like this work i.e a dispatch table. This would make it easier to extend without changing the logic and is shorter
```suggestion
column_to_value_map: dict = {"Exclude": lambda x: self.get_sample_control_status(sample_id=x),
                                    "SampleName": lambda x: self.get_sample_name_from_lims_id(lims_id=x),
                                     "Library_nM": lambda x: self.get_concentrations_from_lims(sample_id=x),
                                     "SequencingDate": lambda x: self.get_sample_sequenced_date(sample_id=x),
                                     sample_project_column_alias: lambda x: self.get_sample_starlims_id(sample_id=x),
                                     }
        for column, value in column_to_value_map.items():
            samplesheet_df = self.add_samplesheet_column(
                samplesheet_df=samplesheet_df,
                new_column=column,
                to_add=samplesheet_df[sample_id_column_alias].apply(
                    value
                ),
            )
```

Reply:
Or is this solution what you meant in the previous comment?